### ISSUE TITLE
Support merging `<excludes/>` elements in child/parent maven projects

### ISSUE THREAD

**Issue by ofirnk** [jOOQ/jOOQ#6556]
##### Parent Maven is:
```xml
	<generator>
		<database>
			<includes>.*</includes>
			<excludes>
				SCHEMA_VERSION
			</excludes>
			<includeExcludeColumns>true</includeExcludeColumns>
		</database>
		<generate>
			.............
		</generate>
		<target>
                        ............
		</target>
	</generator>
```
##### Child Maven is:
```xml
	<generator>
		<database>
			<excludes>
				.*password
			</excludes>
	</generator>
```

### Steps to reproduce the problem:
1. Generate sources
2. There are `getters` etc' generated for `password` db field
==> Child `<excludes/>` rule is ignored

### Versions:

- jOOQ: 3.9.5
- Database (include vendor): PG 9.6
- JDBC Driver (include name if inofficial driver):

------------------------------------------------------------------------

From what I've read about [_maven_ merge strategy](http://blog.sonatype.com/2011/01/maven-how-to-merging-plugin-configuration-in-complex-projects/), such parent-child merging is only possible if `excludes` would be similar to this structure:

##### In parent POM:
```xml
	<generator>
		<database>
			<excludeRules>
				<excludeRule>SCHEMA_VERSION</excludeRule>
			</excludeRules>
	</generator>
```

##### In child POM:
```xml
	<generator>
		<database>
			<excludeRules  combine.children="append">
				<excludeRule>.*password</excludeRule>
			</excludeRules>
	</generator>
```

#### tl;dr
`<excludes>text|pipe|separated</excludes>` -->
##### Support
```xml
      <excludeItems>
          <item>text</item>
          <item>runtime</item>
          <item>separated</item>
      </excludeItems>
```

**Comment by lukaseder** [comment]
Thanks for your suggestion. This is very similar to https://github.com/jOOQ/jOOQ/issues/6550, which we'll certainly consider for jOOQ 3.11. Nice to know that Maven can already do this. We'll need to find a solution that works both with Maven and with the standalone code generation configuration, though.

**Comment by lukaseder** [comment]
Looking into this now. We shouldn't change the existing XML elements for backwards compatibility purposes. People have probably written complicated regular expressions to make sure they include/exclude all the objects they need. Instead, let's offer per-object patterns (https://github.com/jOOQ/jOOQ/issues/5263), and implement those in the way you suggest to allow for merging patterns from parent projects or from `<pluginManagement/>` sections

**Comment by lukaseder** [comment]
Closing this as a duplicate of https://github.com/jOOQ/jOOQ/issues/5263
